### PR TITLE
Ahmedasif/com 217 v3 backend refactor

### PR DESCRIPTION
**Pull request by MAA-Git**
Includes all v3 backend endpoint refactors. Add rename list too. Remove creation of .env files in CI yml since it is redundant.

### CODE REVIEW

File: .github/workflows/desktop_ci.yml
Comment (gapy04) [inline .github/workflows/desktop_ci.yml:1]: I agree with this becoming redundant. Instead, you can have two secrets in Github called AUTH0.env and BACKEND.env. Then you can do:
```
echo ${{ Auth0.env }} > "Auth0.env"
echo ${{ Backend.env }} > "Backend.env"
```
So you only need to change one secret and the code here isn't too big. you'' probably want the env files for the ci/cd overall.

File: src/Backend/Backend.cs [hunk 1]
```diff
@@ -812,36 +810,112 @@ public async Task<bool> MoveIngredient(IUser user, string currListName, string n
...
+            }
+            catch (Exception)
+            {
+                renamed = false;
+            }
+
+            return renamed;
+        }
+
+        private void RenameCacheList(string currListName, string newListName)
```
Comment (gapy04) [inline src/Backend/Backend.cs:906]: Add comment for this function

**Review by gapy04 — CHANGES_REQUESTED** [review]
Add comments and change the ci to include the .env files but in less redundant way

**Comment by MAA-Git** [comment]
**Removing .env files CI:**
Environment variables are not used at build time, only at compile time. Therefore, they are redundant in the CI. AUTH0 env vars will no longer be needed since backend will soon be handling that. Backend env vars to be converted to build configs later in development (once login/logout is handled by backend).

**Review by gapy04 — APPROVED** [review]
Nicely done!

Sounds good about the .env in CI.

**Pull request merged**